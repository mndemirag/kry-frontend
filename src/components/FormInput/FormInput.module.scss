.inputLabel {
  display: flex;
  flex-direction: column;
  font-size: 12px;
  text-transform: uppercase;
  color: rgb(110, 118, 123);
  margin-bottom: 20px;
}

.inputField {
  font-size: 18px;
  background-color: rgb(255, 255, 255);
  padding: 10px 0px;
  border-radius: 0px;
  border-style: none none solid;
  border-bottom-width: 2px;
  border-bottom-color: rgb(206, 208, 210);
  outline: none;

  &:focus-visible {
    border-bottom-color: rgb(10, 126, 133);
  }
}
