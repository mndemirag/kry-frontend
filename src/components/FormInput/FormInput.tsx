import { FormInputProps } from "../../types";

import styles from "./FormInput.module.scss";

export const FormInput: React.FC<FormInputProps> = ({ label, onChange }) => {
  return (
    <label className={styles.inputLabel}>
      {label}
      <input
        type="text"
        className={styles.inputField}
        aria-label={label}
        onChange={(e) => onChange(e.target.value)}
      />
    </label>
  );
};
