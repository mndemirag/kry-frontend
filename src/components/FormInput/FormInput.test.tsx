import { render } from "@testing-library/react";

import { FormInput } from "./FormInput";

const mockChange = jest.fn();

describe("FormInput", () => {
  it("renders properly", () => {
    const { container } = render(
      <FormInput label="input" onChange={mockChange} />
    );
    expect(container).toMatchSnapshot();
  });
});
