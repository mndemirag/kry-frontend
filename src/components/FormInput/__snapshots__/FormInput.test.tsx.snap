// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders properly 1`] = `
<div>
  <label
    class="inputLabel"
  >
    input
    <input
      aria-label="input"
      class="inputField"
      type="text"
    />
  </label>
</div>
`;
